### PR TITLE
Implement EIP-3855: PUSH0 instruction

### PR DESCRIPTION
**Pull request by casiojapi**
## Description

As outlined in [EIP-3855](https://eips.ethereum.org/EIPS/eip-3855), we have introduced a new instruction that pushes the constant value 0 onto the stack. This enhancement reduces contract code size and utilizes a lower gas cost tier (BASE) for this specific operation. This proposal was initially authored by Alex Beregszaszi ([@axic](https://github.com/axic)), Hugo De la cruz ([@hugo-dc](https://github.com/hugo-dc)), Paweł Bylica ([@chfast](https://github.com/chfast)). This implementation aims to optimize gas usage, enhance contract execution efficiency, and improve compatibility with Ethereum.

## Motivation and Context

By implementing the PUSH0 instruction, we address this issue by optimizing the push operation for the frequently used '0' constant. This not only reduces gas costs and contract code size, but also brings us a step closer to better Ethereum compatibility by adopting their standards, part of our commitment to continuously improving the interoperability and performance of the RSKj node.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Comment by casiojapi** [comment]
pipeline:run

File: rskj-core/src/test/java/co/rsk/vm/VMExecutionTest.java [hunk 1]
```diff
@@ -893,4 +893,34 @@ private Program playCodeWithActivationConfig(byte[] code, ActivationConfig.ForBl
 
         return program;
     }
+
+    private void executePush0(String expect, ActivationConfig.ForBlock activations){
```
[the remark this answers is not in the document]
Comment (casiojapi) [reply rskj-core/src/test/java/co/rsk/vm/VMExecutionTest.java:897]: yep, missed that.

File: rskj-core/src/test/java/co/rsk/vm/VMExecutionTest.java [hunk 2]
```diff
@@ -893,4 +893,34 @@ private Program playCodeWithActivationConfig(byte[] code, ActivationConfig.ForBl
...
+                "0000000000000000000000000000000000000000000000000000000000000000",
+                activations);
+    }
+
+    @Test
+    void testPUSH0NoActivation() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(RSKIPYYY)).thenReturn(false);
+
+        Assertions.assertThrows(Program.IllegalOperationException.class, () -> {
```
[the remark this answers is not in the document]
Comment (casiojapi) [reply rskj-core/src/test/java/co/rsk/vm/VMExecutionTest.java:919]: :)

**Comment by casiojapi** [comment]
pipeline:run

**Review by asoto-iov — APPROVED** [review]

File: rskj-core/src/main/java/org/ethereum/config/blockchain/upgrades/ConsensusRule.java [hunk 3]
```diff
@@ -86,6 +86,7 @@ public enum ConsensusRule {
     RSKIP377("rskip377"),
     RSKIP383("rskip383"),
     RSKIP385("rskip385"),
+    RSKIP389("rskip389"),
```
Comment (Vovchyk) [inline rskj-core/src/main/java/org/ethereum/config/blockchain/upgrades/ConsensusRule.java:89]:
```suggestion
    RSKIP389("rskip398"),
```
typo

Comment (casiojapi) [reply rskj-core/src/main/java/org/ethereum/config/blockchain/upgrades/ConsensusRule.java:89]: oh damn, thanks for spotting that

Comment (casiojapi) [reply rskj-core/src/main/java/org/ethereum/config/blockchain/upgrades/ConsensusRule.java]: updated it

**Comment by casiojapi** [comment]
pipeline:run

**Comment by casiojapi** [comment]
pipeline:run

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**